src/components/elements/metadataKeyValue/MetadataKeyValuePair.tsx

```diff
@@ -0,0 +1,147 @@
+  saveValueLocally?: (index: number, newVal?: string | undefined) => void;
+  saveValueTypeLocally?: (index: number, newVal?: string | number | undefined) => void;
+  deleteFieldPermanently?: (index: number) => void;
+}
+
+export function MetadataKeyValuePair({
+  index,
+  keyName,
+  value,
+  valueType,
+  isEditable,
+  isAutoGenerated,
+  isNewlyCreated,
+  deletionEnabled,
+  saveKeyNameLocally,
+  saveValueLocally,
+  saveValueTypeLocally,
+  deleteFieldPermanently
+}: MetadataKeyValuePairProps) {
+  const valueTypes: RadioButtonItemProps[] = [
+    {
+      id: 'INTEGER',
+      label: 'Int'
+    },
+    {
+      id: 'FLOAT',
+      label: 'Float'
+    },
+    {
+      id: 'BOOLEAN',
+      label: 'Boolean'
+    },
+    {
+      id: 'STRING',
+      label: 'String'
+    },
+    {
+      id: 'BLOB',
+      label: 'Blob'
+    }
+  ];
+
+  return (
+    <Box
+      sx={{
+        display: 'flex',
+        flexDirection: 'row',
+        width: '100%',
+        height: '100%',
+        backgroundColor: '#F8FAFC',
+        borderBottom: '1px solid #E2E8F0',
+        alignItems: 'center',
+        gap: '8px'
+      }}
+    >
+      <Box
+        sx={{
+          display: 'flex',
+          width: '35%',
+          height: '100%',
+          flexShrink: 0
+        }}
+      >
+        {/*key name should not be editable unless its newly created*/}
+        <CustomTextField
+          readOnly={!isNewlyCreated}
+          value={keyName}
+          saveLocallyHandler={(newVal) => {
+            if (saveKeyNameLocally) {
+              saveKeyNameLocally(index, newVal);
+            }
+          }}
+          helperText={isAutoGenerated ? 'Auto-generated' : undefined}
+          placeholder={'Enter field name'}
+        />
+      </Box>
+      <Box
+        sx={{
+          display: 'flex',
+          width: '65%',
+          height: '100%'
+        }}
+      >
+        {isNewlyCreated && (
+          <Box flexShrink={0}>
+            <DropdownV2
+              onChange={(event, value) => {
+                if (saveValueTypeLocally) {
+                  saveValueTypeLocally(index, value?.id);
+                }
+              }}
+              options={valueTypes}
+              isReadOnly={true}
+              label={'Value type'}
+              errored={false}
+              maxWidth={'130px'}
+              height={'36px'}
+              isSquareCorners={true}
+              withoutBorder={true}
+              backgroundColorFocus={'white'}
+            />
+          </Box>
+        )}
+        <CustomTextField
+          readOnly={!isEditable || !!isAutoGenerated}
+          value={value}
+          saveLocallyHandler={(newVal) => {
+            if (saveValueLocally) {
+              saveValueLocally(index, newVal);
+            }
+          }}
+          placeholder={isNewlyCreated ? 'Add value' : 'Typing'}
+        />
+      </Box>
+      {isEditable && deletionEnabled && !isAutoGenerated && (
```

Comment:
I think the second one should be named `isRemovable` to be consistent with `isEditable` and also to make it generic. In addition `isAutoGenerated` shouldn't be a thing here, the values of `isEditable` and `isRemovable` should both be false for an item that is auto-generated. That predicating whether the field is editable is decided by the caller component.